src/games/strategy/triplea/ai/proAI/util/ProBattleUtils.java

```diff
@@ -38,9 +37,13 @@
   public final static int SHORT_RANGE = 2;
   public final static int MEDIUM_RANGE = 3;
 
-  private final static IOddsCalculator calc = new ConcurrentOddsCalculator("ProAI");
+  private static IOddsCalculator calc;
   private static boolean isCanceled = false;
 
+  public static void setOddsCalculator(final IOddsCalculator calc) {
```

Comment:
Static methods that control state are effectively by definition procedural. This is error prone since whenever you use ProBattleUtils, you need to be aware of *EVERY* usage of the class, otherwise you do not know what state this class would be in. Extract this functionality into a stateful class.  SRP and encapsulation violation here.